src/main/kotlin/org/ozwillo/dcimporter/web/MarchePublicHandler.kt

```diff
@@ -477,6 +488,284 @@ class MarchePublicHandler(private val datacoreProperties: DatacoreProperties,
+                                    val entrepriseIri = "FR/" + registre.getStringValue("mpretrait:entreprise").substringAfterLast("/")
+
+                                    val registreRetrait = RegistreRetrait.fromDCObject(registre)
+
+                                    if (!personneIri.isEmpty()){
+                                        val dcPersonne = datacoreService.getResourceFromIRI(MP_PROJECT, MSUtils.PERSONNE_TYPE, personneIri, bearer)
+                                        registreRetrait.personne = Personne.fromDCObject(dcPersonne)
+                                    }
+                                    if (entrepriseIri != "FR/"){
+                                        val dcEntreprise = datacoreService.getResourceFromIRI(MP_PROJECT, ORG_TYPE, entrepriseIri, bearer)
+                                        registreRetrait.entreprise = Organization.fromDcObject(dcEntreprise)
+                                    }
+
+                                    body.add(registreRetrait)
+                                }
+                            }
+                        }
+                        ok().contentType(MediaType.APPLICATION_JSON).body(BodyInserters.fromObject(body))
+                    }
+        }catch (e: HttpClientErrorException){
+            val body = when(e.statusCode) {
+                HttpStatus.UNAUTHORIZED -> "Token unauthorized, maybe it is expired ?"
+                HttpStatus.NOT_FOUND -> "Consultation with reference ${req.pathVariable("reference")} does not exist"
+                else -> "Unexpected error"
+            }
+            status(e.statusCode).body(BodyInserters.fromObject(body))
+        }
+
+    }
+
+    fun getRegistreRetraitResumeForConsultation(req: ServerRequest): Mono<ServerResponse>{
+        val bearer = extractBearer(req.headers())
+        val siret = req.pathVariable("siret")
+        val iri = "FR/$siret/${req.pathVariable("reference")}"
+        val type = MSUtils.RETRAIT_TYPE
+
+        val startParam = if (req.queryParam("start").isPresent) req.queryParam("start").get().toInt() else 0
+        val maxParam = if (req.queryParam("limit").isPresent) req.queryParam("limit").get().toInt() else 50
+
+        try {
+            datacoreService.getResourceFromIRI(MP_PROJECT, ORG_TYPE, "FR/$siret", bearer)
+        } catch (e: HttpClientErrorException) {
+            val body = when(e.statusCode) {
+                HttpStatus.UNAUTHORIZED -> "Token unauthorized, maybe it is expired ?"
+                HttpStatus.NOT_FOUND -> "Organization with SIRET $siret does not exist"
+                else -> "Unexpected error"
+            }
+
+            return status(e.statusCode).body(BodyInserters.fromObject(body))
+        }
+
+        return try {
+            val dcConsultation = datacoreService.getResourceFromIRI(MP_PROJECT, CONSULTATION_TYPE, iri, bearer)
+            val personnes: MutableList<String> = arrayListOf()
```

Comment:
I'm not convinced.

If we list all the persons and organizations in the summary, well, this is no more really a summary.

Reply:
The idea here is to list a resume of "retrait" by organization while the detailed "retrait" list list all "retrait", each with the concerned organization.
Some improvement have been made on this method for the same result : a list of organization and contact with number of "retraits" done, date of first and date of last one.